zos_basics/constructs/zos_job_submit_ansible_constructs.yaml

```diff
@@ -0,0 +1,585 @@
+#   - mode_verbose          - run playbook with added verbosity
+#
+# Usage:
+#  ansible-playbook -i <inventory> <playbook> --tags "....."
+#
+#   1) Run in development mode using JSON cached input
+#         ansible-playbook -i inventory zos_job_submit_ansible_constructs.yaml --tags "mode_development_json"
+#   2) Run in development mode using YAML cached input
+#         ansible-playbook -i inventory zos_job_submit_ansible_constructs.yaml --tags "mode_development_yaml"
+#   3) Run in development mode using JSON cached input with verbose
+#         ansible-playbook -i inventory zos_job_submit_ansible_constructs.yaml --tags "mode_development_json,mode_verbose"
+#   4) Run in development mode using YAML cached input with verbose
+#         ansible-playbook -i inventory zos_job_submit_ansible_constructs.yaml --tags "mode_development_yaml,mode_verbose"
+#   5) Run in production mode on z/OS target
+#         ansible-playbook -i inventory zos_job_submit_ansible_constructs.yaml --tags "mode_production"
+#   6) Run in production mode on z/OS target with verbose
+#         ansible-playbook -i inventory zos_job_submit_ansible_constructs.yaml --tags "mode_production,mode_verbose"
+#
+# Requirements:
+#   1) IBM® z/OS® core collection 1.0.0 or later
+#   2) Ansible 2.10 or later
+#   3) jmespath (`pip install jmespath`)
+#
+# Notes:
+#   * If you do not run with Ansible 2.10 or later, you will encounter an
+#     error similar to this:
+#       "Invalid data passed to 'loop', it requires a list, got this
+#        instead: {{ submit_job_list | community.general.json_query('ddnames[*].ddname') }}.
+#        Hint: If you passed a list/dict of just one element, try adding wantlist=True
+#        to your lookup invocation or use q/query instead of lookup."}""
+#   * If you do not install `jmespath` you will encounter an error on tasks that
+#     use `json_query` (community.general.json_query) similar to this:
+#       "You need to install \"jmespath\" prior to running json_query filter"}"
+#
+# Documentation and further reading:
+#   Review the release notes for the collection version dependencies here:
+#     https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/release_notes.html
+#   Install `jmespath`
+#     https://pypi.org/project/jmespath/
+#
+###############################################################################
+
+---
+- hosts: zos_host
+  collections:
+    - ibm.ibm_zos_core
+  gather_facts: no
+  environment: "{{ environment_vars }}"
+  vars:
+    ############################################################################
+    # YAML representation of the zos_job_submit response used for development to
+    # simulate a job submission used to expedite development. Allows for
+    # playbook development to be done without actually submitting jobs when
+    # --tags "mode_development_yaml"
+    ############################################################################
+    job_sub_result_yaml:
```

Comment:
I do like the idea and had thought the same but because I have sent this playbook to users as an email attachment with the goal being that is a beginner sample that I should keep it all inline so they can see the content and not run into any setup issues. I could go either way on this.

Reply:
fair point. sounds good to me.